### PR TITLE
Adding Symfony 4 support

### PR DESCRIPTION
**Pull request by hermann8u**
The PageRepository and CategoryRepository are now explicitly declared as public to allow compatibility with Symfony 4

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ composer.lock
 vendor/
 build/
 phpunit.xml
+.idea/
```
Comment (Pierstoval) [inline .gitignore]: This should never be added in a project, and instead be added to your global git configuration

File: Repository/AbstractCmsRepository.php [hunk 2]
```diff
@@ -18,7 +18,7 @@
  *
  * @author Sandor Farkas <farkas.berlin@gmail.com>
  */
-class AbstractCmsRepository extends EntityRepository
+abstract class AbstractCmsRepository extends EntityRepository
```
Comment (Pierstoval) [inline Repository/AbstractCmsRepository.php]: There's no need to make the class abstract for now, and it's technically a BC break

**Review by Pierstoval — CHANGES_REQUESTED** [review]

File: Repository/AbstractCmsRepository.php (hunk 2 above)
Comment (hermann8u) [reply Repository/AbstractCmsRepository.php]: I was just thinking that you forgot to put the class abstract and I would to correct it. I will remove this.

File: .gitignore [hunk 3]
```diff
@@ -2,4 +2,4 @@ composer.phar
 composer.lock
 vendor/
 build/
-phpunit.xml
+phpunit.xml
```
Comment (Pierstoval) [inline .gitignore]: Please keep the newline at the end of the file

**Review by Pierstoval — CHANGES_REQUESTED** [review]

**Comment by Pierstoval** [comment]
Thanks @hermann8u

**Pull request merged**